src/libs/actions/connections/SageIntacct.ts

```diff
@@ -14,4 +28,296 @@ function connectToSageIntacct(policyID: string, credentials: SageIntacctCredenti
+                    intacct: {
+                        config: {
+                            pendingFields: {
+                                tax: null,
+                            },
+                            errorFields: {
+                                tax: undefined,
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+    API.write(WRITE_COMMANDS.UPDATE_SAGE_INTACCT_SYNC_TAX_CONFIGURATION, {policyID, enabled}, {optimisticData, failureData, successData});
+}
+
+function prepareOnyxDataForUserDimensionUpdate(policyID: string, dimensionName: string, newDimensions: SageIntacctDimension[]) {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    intacct: {
+                        config: {
+                            mappings: {
+                                dimensions: newDimensions,
+                            },
+                            pendingFields: {[`dimension_${dimensionName}`]: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE},
+                            errorFields: {[`dimension_${dimensionName}`]: null},
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    intacct: {
+                        config: {
+                            mappings: {
+                                dimensions: newDimensions,
+                            },
+                            pendingFields: {[`dimension_${dimensionName}`]: null},
+                            errorFields: {[`dimension_${dimensionName}`]: ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('common.genericErrorMessage')},
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                connections: {
+                    intacct: {
+                        config: {
+                            mappings: {
+                                dimensions: newDimensions,
+                            },
+                            pendingFields: {[`dimension_${dimensionName}`]: null},
+                            errorFields: {[`dimension_${dimensionName}`]: null},
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    return {optimisticData, failureData, successData};
+}
+
+function addSageIntacctUserDimensions(
```

Comment:
cc @NikkiWines on this function to make sure your API lines up with what's being sent here and below, or if @SzymczakJ needs to tweak things.

Reply:
Sounds like you guys have this one sorted based on [this discussion](https://expensify.slack.com/archives/C06ML6X0W9L/p1720469871270959?thread_ts=1719447487.162559&cid=C06ML6X0W9L)